L1Trigger/Phase2L1ParticleFlow/test/testOutputFiles.cpp

```diff
@@ -0,0 +1,611 @@
+    std::cerr << "ERROR::testDumpFile Filename 3 must be a COE (.coe) file" << std::endl << usage.str() << std::endl;
+    return -6;
+  }
+
+  // report the program configuraion
+  std::cout << "Configuration:" << std::endl
+            << "==============" << std::endl
+            << "Number of tests (events): " << NTEST << std::endl
+            << "Report every N tests: " << REPORT_EVERY_N << std::endl
+            << "Number of regions (in eta-phi): " << regions_.size() << std::endl;
+  for (unsigned int iregion = 0; iregion < regions_.size(); iregion++) {
+    printf("\t%i : eta=[%0.4f,%0.4f] phi=[%0.4f,%0.4f]\n",
+           iregion,
+           regions_[iregion].etaMin,
+           regions_[iregion].etaMax,
+           regions_[iregion].phiCenter - regions_[iregion].phiHalfWidth,
+           regions_[iregion].phiCenter + regions_[iregion].phiHalfWidth);
+  }
+  std::cout << "Number of tracks per sector: " << NTRACKS_PER_SECTOR << std::endl
+            << "Number of bits per track: " << NBITS_PER_TRACK << std::endl
+            << "==============" << std::endl
+            << std::endl;
+
+  // open the files for testing
+  TFile *rfile_ = TFile::Open(filename_root.c_str(), "READ");
+  if (!rfile_) {
+    std::cerr << "ERROR::testDumpFile Cannot open '" << filename_root << "'" << std::endl;
+    return -7;
+  }
+  fwlite::Event rfileentry_(rfile_);
+  FILE *dfile_(fopen(filename_dump.c_str(), "rb"));
+  if (!dfile_) {
+    std::cerr << "ERROR::testDumpFile Cannot read '" << filename_dump << "'" << std::endl;
+    return -8;
+  }
+  std::ifstream cfile_(filename_coe);
+  if (!cfile_) {
+    std::cerr << "ERROR::testDumpFile Cannot read '" << filename_coe << "'" << std::endl;
+    return -9;
+  }
+  GotoLine(cfile_, 11);  //Skip the header of the COE file
+
+  TrackMap tracks_root, tracks_dump, tracks_coe;
+
+  // run the tests for multiple events
+  for (int test = 1; test <= NTEST; ++test) {
+    if (test % REPORT_EVERY_N == 1)
+      std::cout << "Doing test " << test << " ... " << std::endl;
+
+    tracks_root = get_tracks_from_root_file(rfileentry_, test - 1, test == 1);
+    tracks_dump = get_tracks_from_dump_file(dfile_, test == 1);
+    tracks_coe = get_tracks_from_coe_file(cfile_, test == 1);
+
+    if (test % REPORT_EVERY_N == 1)
+      std::cout << "Comparing the ROOT tracks to the dump tracks in event " << test << " ... " << std::flush;
+    if (!compare_maps(tracks_root, tracks_dump))
+      return -10;
+    if (test % REPORT_EVERY_N == 1)
+      std::cout << "DONE" << std::endl;
+
+    if (test % REPORT_EVERY_N == 1)
+      std::cout << "Comparing the ROOT tracks to the coe tracks in event " << test << " ... " << std::flush;
+    if (!compare_maps(tracks_root, tracks_coe))
+      return -11;
+    if (test % REPORT_EVERY_N == 1)
+      std::cout << "DONE" << std::endl << std::endl;
+  }
+
+  std::cout << std::endl << "The dump and coe outputs match the ROOT outputs for all events!" << std::endl;
+  return 0;
+}
+
+/*
+USE:
```

Comment:
this should not be necessary if scram knows how to compile the file automatically